src/structurizr/view/view_set.py

```diff
@@ -75,25 +77,85 @@ class ViewSet(ModelRefMixin, AbstractBase):
     def __init__(
         self,
         *,
-        model: Optional["Model"] = None,
+        model: Optional["Model"],
         # TODO:
         # enterprise_context_views: Iterable[EnterpriseContextView] = (),
         system_landscape_views: Iterable[SystemLandscapeView] = (),
         system_context_views: Iterable[SystemContextView] = (),
         container_views: Iterable[ContainerView] = (),
+        component_views: Iterable[ComponentView] = (),
         configuration: Optional[Configuration] = None,
         **kwargs
     ) -> None:
         """Initialize a view set."""
         super().__init__(**kwargs)
         # self.enterprise_context_views = set(enterprise_context_views)
-        self.system_landscape_views = set(system_landscape_views)
-        self.system_context_views = set(system_context_views)
-        self.container_views = set(container_views)
+        self.system_landscape_views: Iterable[SystemLandscapeView] = set(
+            system_landscape_views
+        )
+        self.system_context_views: Iterable[SystemContextView] = set(
+            system_context_views
+        )
+        self.container_views: Iterable[ContainerView] = set(container_views)
+        self.component_views: Iterable[ComponentView] = set(component_views)
         # TODO
         self.configuration = Configuration() if configuration is None else configuration
         self.set_model(model)
 
+    @classmethod
+    def hydrate(cls, views: ViewSetIO, model: "Model") -> "ViewSet":
+        system_landscape_views = []
```

Comment:
I'd suggest to create a separate method for each logical block here.

```suggestion
        system_landscape_views = self._hydrate_system_landscape_views(...)
```

and so on for the others.